exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/metric_exporter.rb

```diff
@@ -0,0 +1,287 @@
+          # this exporter is registered to a {OpenTelemetry::SDK::Metric::MeterProvider}
+          # object.
+          #
+          # @param [optional Numeric] timeout An optional timeout in seconds.
+          def force_flush(timeout: nil)
+            SUCCESS
+          end
+
+          # Called when {OpenTelemetry::SDK::Metric::MeterProvider#shutdown} is called, if
+          # this exporter is registered to a {OpenTelemetry::SDK::Trace::MeterProvider}
+          # object.
+          #
+          # @param [optional Numeric] timeout An optional timeout in seconds.
+          def shutdown(timeout: nil)
+            @shutdown = true
+            @http.finish if @http.started?
+            SUCCESS
+          end
+
+          private
+
+          def fetch_ssl_verify_mode
+            if ENV.key?('OTEL_RUBY_EXPORTER_OTLP_SSL_VERIFY_PEER')
+              OpenSSL::SSL::VERIFY_PEER
+            elsif ENV.key?('OTEL_RUBY_EXPORTER_OTLP_SSL_VERIFY_NONE')
+              OpenSSL::SSL::VERIFY_NONE
+            else
+              OpenSSL::SSL::VERIFY_PEER
+            end
+          end
+
+          def http_connection(uri, ssl_verify_mode, certificate_file)
+            http = Net::HTTP.new(uri.host, uri.port)
+            http.use_ssl = uri.scheme == 'https'
+            http.verify_mode = ssl_verify_mode
+            http.ca_file = certificate_file unless certificate_file.nil?
+            http.keep_alive_timeout = KEEP_ALIVE_TIMEOUT
+            http
+          end
+
+          # The around_request is a private method that provides an extension
+          # point for the exporters network calls. The default behaviour
+          # is to not trace these operations.
+          #
+          # An example use case would be to prepend a patch, or extend this class
+          # and override this method's behaviour to explicitly trace the HTTP request.
+          # This would allow you to trace your export pipeline.
+          def around_request
+            OpenTelemetry::Common::Utilities.untraced { yield }
+          end
+
+          def send_bytes(bytes, timeout:) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+            return FAILURE if bytes.nil?
+
+            retry_count = 0
+            timeout ||= @timeout
+            start_time = OpenTelemetry::Common::Utilities.timeout_timestamp
+            around_request do # rubocop:disable Metrics/BlockLength
+              request = Net::HTTP::Post.new(@path)
+              request.body = if @compression == 'gzip'
+                               request.add_field('Content-Encoding', 'gzip')
+                               Zlib.gzip(bytes)
+                             else
+                               bytes
+                             end
+              request.add_field('Content-Type', 'application/x-protobuf')
+              @headers.each { |key, value| request.add_field(key, value) }
```

Comment:
We should hoist all this outside of the `around_request` block.

Reply:
Yup, this is now out of date with the released `opentelemetry-exporter-otlp` gem. There's a bit of work to be done here in general.

Reply:
Removed from the PR.